File: VB/Dashboard_Coloring/Form1.vb
Imports System.Data
Imports System.Drawing
Imports DevExpress.DashboardCommon
Imports DevExpress.Drawing
Imports DevExpress.XtraBars.Ribbon

Namespace Dashboard_Coloring
	Partial Public Class Form1
		Inherits RibbonForm

		Public Sub New()
			InitializeComponent()
			dashboardDesigner1.CreateRibbon()
			' Create a color table that contains dimension values, measures and colors.
			Dim colorTable As DataTable = CreateColorTable()
			' Load a dashboard from the XML file.
			Dim dashboard As New Dashboard()
			dashboard.LoadFromXml("..\..\Data\Dashboard.xml")
			Dim dataSource As IDashboardDataSource = dashboard.DataSources("dataSource1")

			' Specify the coloring mode for the pie series, pie measures and chart argument.
			Dim pie1 As PieDashboardItem = CType(dashboard.Items("pieDashboardItem1"), PieDashboardItem)
			Dim chart1 As ChartDashboardItem = CType(dashboard.Items("chartDashboardItem1"), ChartDashboardItem)
			pie1.SeriesDimensions(0).ColoringMode = ColoringMode.Hue
			pie1.ColoringOptions.MeasuresColoringMode = ColoringMode.Hue
			chart1.Arguments(0).ColoringMode = ColoringMode.Hue

			For Each row As DataRow In colorTable.Rows
				dashboard.ColorScheme.Add(CreateColorSchemeEntry(row, dataSource, False))
				dashboard.ColorScheme.Add(CreateColorSchemeEntry(row, dataSource, True))
			Next row
			dashboardDesigner1.Dashboard = dashboard
		End Sub

		' Create color scheme entries to map dimension values, measures and colors.
		Private Function CreateColorSchemeEntry(ByVal colorSchemeRecord As DataRow, ByVal dataSource As IDashboardDataSource, ByVal includeMeasures As Boolean) As ColorSchemeEntry
			Dim categoryDefinition As New DimensionDefinition("CategoryName")
			Dim countryDefinition As New DimensionDefinition("Country")
			Dim priceDefinition As New MeasureDefinition("Extended Price")

			Dim entry As New ColorSchemeEntry()
			entry.DimensionKeys.Add(New ColorSchemeDimensionKey(categoryDefinition, colorSchemeRecord("CategoryName")))
			entry.DimensionKeys.Add(New ColorSchemeDimensionKey(countryDefinition, colorSchemeRecord("Country")))
			If includeMeasures Then
				entry.MeasureKey = New ColorSchemeMeasureKey(priceDefinition)
			End If
			' Use a DashboardPaletteItem to define the color, a fill style (hatch), and line style.
			entry.ColorDefinition = New ColorDefinition(New DashboardPaletteItem(DirectCast(colorSchemeRecord("color"), Color), DirectCast(colorSchemeRecord("hatch"), DXHatchStyle), DirectCast(colorSchemeRecord("line"), DXDashStyle)))
			entry.DataSource = dataSource
			Return entry
		End Function

		Private Function CreateColorTable() As DataTable
			Dim colorTable As New DataTable()
			colorTable.Columns.Add("CategoryName", GetType(String))
			colorTable.Columns.Add("Country", GetType(String))
			colorTable.Columns.Add("measure", GetType(String))
			colorTable.Columns.Add("color", GetType(Color))
			colorTable.Columns.Add("hatch", GetType(DXHatchStyle))
			colorTable.Columns.Add("line", GetType(DXDashStyle))
			colorTable.Rows.Add("Beverages", "UK", "Extended Price", Color.Red, DXHatchStyle.DiagonalCross, DXDashStyle.Dash)
			colorTable.Rows.Add("Beverages", "USA", "Extended Price", Color.Green, DXHatchStyle.Sphere, DXDashStyle.Dot)
			Return colorTable
		End Function
	End Class
End Namespace

File: CS/Dashboard_Coloring/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.Drawing;
using DevExpress.XtraBars.Ribbon;
using System.Data;
using System.Drawing;

namespace Dashboard_Coloring {
    public partial class Form1 : RibbonForm {
        public Form1() {
            InitializeComponent();
            dashboardDesigner1.CreateRibbon();

            // Create a color table that contains dimension values, measures and colors.
            DataTable colorTable = CreateColorTable();
            // Load a dashboard from the XML file.
            Dashboard dashboard = new Dashboard(); dashboard.LoadFromXml(@"..\..\Data\Dashboard.xml");
            IDashboardDataSource dataSource = dashboard.DataSources["dataSource1"];

            // Specify the coloring mode for the pie series, pie measures and chart argument.
            PieDashboardItem pie1 = (PieDashboardItem)dashboard.Items["pieDashboardItem1"];
            ChartDashboardItem chart1 = (ChartDashboardItem)dashboard.Items["chartDashboardItem1"];
            pie1.SeriesDimensions[0].ColoringMode = ColoringMode.Hue;
            pie1.ColoringOptions.MeasuresColoringMode = ColoringMode.Hue;
            chart1.Arguments[0].ColoringMode = ColoringMode.Hue;

            foreach (DataRow row in colorTable.Rows) {
                dashboard.ColorScheme.Add(CreateColorSchemeEntry(row, dataSource, false));
                dashboard.ColorScheme.Add(CreateColorSchemeEntry(row, dataSource, true));
            }
            dashboardDesigner1.Dashboard = dashboard;
        }

        // Create color scheme entries to map dimension values, measures and colors.
        private ColorSchemeEntry CreateColorSchemeEntry(DataRow colorSchemeRecord,
                                                        IDashboardDataSource dataSource,
                                                        bool includeMeasures) {
            DimensionDefinition categoryDefinition = new DimensionDefinition("CategoryName");
            DimensionDefinition countryDefinition = new DimensionDefinition("Country");
            MeasureDefinition priceDefinition = new MeasureDefinition("Extended Price");

            ColorSchemeEntry entry = new ColorSchemeEntry();
            entry.DimensionKeys.Add(new ColorSchemeDimensionKey(categoryDefinition,
                colorSchemeRecord["CategoryName"]));
            entry.DimensionKeys.Add(new ColorSchemeDimensionKey(countryDefinition,
                colorSchemeRecord["Country"]));
            if (includeMeasures)
                entry.MeasureKey = new ColorSchemeMeasureKey(priceDefinition);
            // Use a DashboardPaletteItem to define the color, a fill style (hatch), and line style.
            entry.ColorDefinition = new ColorDefinition(new DashboardPaletteItem(
                (Color)colorSchemeRecord["color"],
                (DXHatchStyle)colorSchemeRecord["hatch"],
                (DXDashStyle)colorSchemeRecord["line"]));
            entry.DataSource = dataSource;
            return entry;
        }

        private DataTable CreateColorTable() {
            DataTable colorTable = new DataTable();
            colorTable.Columns.Add("CategoryName", typeof(string));
            colorTable.Columns.Add("Country", typeof(string));
            colorTable.Columns.Add("measure", typeof(string));
            colorTable.Columns.Add("color", typeof(Color));
            colorTable.Columns.Add("hatch", typeof(DXHatchStyle));
            colorTable.Columns.Add("line", typeof(DXDashStyle));
            colorTable.Rows.Add("Beverages", "UK", "Extended Price", Color.Red, DXHatchStyle.DiagonalCross, DXDashStyle.Dash);
            colorTable.Rows.Add("Beverages", "USA", "Extended Price", Color.Green, DXHatchStyle.Sphere, DXDashStyle.Dot);
            return colorTable;
        }
    }
}